runtime/handler.go

```diff
@@ -64,12 +65,18 @@ func ForwardResponseStream(ctx context.Context, mux *ServeMux, marshaler Marshal
 		case resp == nil:
 			buf, err = marshaler.Marshal(errorChunk(status.New(codes.Internal, "empty response")))
 		default:
-			result := map[string]interface{}{"result": resp}
-			if rb, ok := resp.(responseBody); ok {
-				result["result"] = rb.XXX_ResponseBody()
+			if httpBody, ok := resp.(*httpbody.HttpBody); ok {
+				buf = httpBody.GetData()
+				w.Header().Set("Content-Type", httpBody.GetContentType())
+			} else {
+				result := map[string]interface{}{"result": resp}
+				if rb, ok := resp.(responseBody); ok {
+					result["result"] = rb.XXX_ResponseBody()
+				}
+
+				buf, err = marshaler.Marshal(result)
```

Comment:
Could you merge this if-else into the encapsulating switch?

Reply:
As discussed, resolving the review comment.

Reply:
I wonder if this would work:
```go
var buf []byte
httpBody, isHttpBody := resp.(*httpbody.HttpBody)
switch {
case resp == nil:
	buf, err = marshaler.Marshal(errorChunk(status.New(codes.Internal, "empty response")))
case isHttpBody:
	buf = httpBody.GetData()
	w.Header().Set("Content-Type", httpBody.GetContentType())
default:
	result := map[string]interface{}{"result": resp}
	if rb, ok := resp.(responseBody); ok {
		result["result"] = rb.XXX_ResponseBody()
	}
	buf, err = marshaler.Marshal(result)
}
```
?

I'd like to avoid the `if-else`, it's a bit messy to me. What do you think?